lib/app/features/auth/providers/onboarding_data_provider.dart

```diff
@@ -0,0 +1,187 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:ice/app/features/auth/providers/auth_provider.dart';
+import 'package:ice/app/features/nostr/providers/nostr_keystore_provider.dart';
+import 'package:ice/app/features/user/model/user_delegation.dart';
+import 'package:ice/app/features/user/model/user_identity.dart';
+import 'package:ice/app/features/user/model/user_metadata.dart';
+import 'package:ice/app/features/user/model/user_relays.dart';
+import 'package:ice/app/features/user/providers/current_user_identity_provider.dart';
+import 'package:ice/app/features/user/providers/user_delegation_provider.dart';
+import 'package:ice/app/features/user/providers/user_metadata_provider.dart';
+import 'package:ice/app/features/user/providers/user_relays_provider.dart';
+import 'package:ice/app/features/wallets/providers/main_wallet_provider.dart';
+import 'package:nostr_dart/nostr_dart.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'onboarding_data_provider.freezed.dart';
+part 'onboarding_data_provider.g.dart';
+
+@Freezed(copyWith: true, equal: true)
+class OnboardingState with _$OnboardingState {
+  const factory OnboardingState({
+    String? name,
+    String? displayName,
+    List<String>? languages,
+    List<String>? followees,
+  }) = _OnboardingState;
+}
+
+@Riverpod(keepAlive: true)
+class OnboardingData extends _$OnboardingData {
```

Comment:
This class is waaay too large, and gonna be even larger. Consider splitting it into smaller ones.